src/charts/stacked-area.js

```diff
@@ -1019,11 +1019,14 @@ define(function(require){
             cleanDataPointHighlights();
 
             // ensure order stays constant
-            values = values
-                .filter(v => !!v)
-                .sort((a,b) => order.indexOf(a.name) > order.indexOf(b.name));
+            let sortedValues = [];
+            values = values.filter(v => !!v);
 
-            values.forEach((d, index) => {
+            for(let i = 0; i< order.length; i++){
+                sortedValues.push(values.find(o=> o.name === order[i]));
+            }
+
```

Comment:
I would do something like:
```
let sortedValues = order.reduce((acc, current) => { 
    return [...acc, values.find(({name}) => name === current))]
},[]);
```
Or similar, I haven't tested it.

Reply:
any idea why the js sort does that with more than 10 areas?

I made a separate codepen to only demonstrate the odd behavior of the sort.
https://codepen.io/flacoman91/pen/gBZKrz?editors=0010

Reply:
Interesting! No idea though...